pkg/tracing/provider/factory.go

```diff
@@ -0,0 +1,50 @@
+package provider
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/thanos/pkg/tracing/provider/jaeger"
+	"github.com/improbable-eng/thanos/pkg/tracing/provider/stackdriver"
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+type TracingProvider string
+
+const (
+	STACKDRIVER TracingProvider = "STACKDRIVER"
+	JAEGER      TracingProvider = "JAEGER"
+)
+
+type TracingConfig struct {
+	Type   TracingProvider `yaml:"type"`
```

Comment:
<3

Can we add `sampling` here? I think sampling is common across providers? (I might be wrong?) Not sure - not a blocker

Reply:
jaeger has remote sampling parameter, therefore it's not universal parameter.

Reply:
Wait - Could we add client one as wel?. To avoid sending traces we don't care if sampling is less than 100%? We can do that in separate PR.

Reply:
I think, that we can add sampling here.
Sorry, I don't understand you ) What do you mean by `cleint one`? What about "if sampling is less than 100%"?

Reply:
Maybe you want "clientSamplingParam"?)

Reply:
Sampling means "every 'sampling' request emits trace", so 100% sampling means literally sampling = 1 so every request emits trace. Client means before sending trace to backend we check what is sampling and how many of requests should send it. This is what sampling for stackdriver tracing means. I think it should be part of common configuration, but we could add it in per provider config either (: not a blocker.

Reply:
Ignore me. I can see jager has things like 
```
SamplerType            string `yaml:"sampler_type"`
	SamplerParam           string `yaml:"sampler_param"`
	SamplerManagerHostPort string `yaml:"sampler_manager_host_port"`
	SamplerMaxOperations   string `yaml:"sampler_max_operations"`
	SamplerRefreshInterval string `yaml:"sampler_refresh_interval"`
```
So it has sampling but configured in different way. Let's keep sampling configuration inside Config interface then :+1: